kore/src/Kore/Reachability/Claim.hs

```diff
@@ -655,6 +662,158 @@ checkImplicationWorker (ClaimPattern.refreshExistentials -> claimPattern) =
+
+When the implication formula is not valid, the function behaves
+similar to @checkImplicationWorker@:
+
+* If unification @⌈t(X) ∧ t'(X, Y)⌉@ succeeded, the return value is
+   'NotImpliedStuck' with the refuted term,
+* otherwise, the function returns 'NotImplied'.
+-}
+checkSimpleImplication ::
+    forall m.
+    (MonadSimplify m) =>
+    Pattern RewritingVariableName -> -- left
+    Pattern RewritingVariableName -> -- right
+    [ElementVariable RewritingVariableName] -> -- existentials
+    m (CheckImplicationResult (ClaimPattern, Maybe (Substitution RewritingVariableName)))
+checkSimpleImplication inLeft inRight inExistentials =
+    do
+        assertFunctionLikeConfiguration claimPattern
+        let definedConfig =
+                Pattern.andCondition renamedLeft $
+                    from $ makeCeilPredicate leftTerm
+        trivial <-
+            fmap isBottom $
+                (liftSimplifier . SMT.Evaluator.filterMultiOr)
+                    =<< Pattern.simplify definedConfig
+
+        if trivial
+            then pure $ Implied (claimPattern, Just mempty)
+            else do
+                -- attempt term unification (to remember the substitution
+                unified <-
+                    Logic.observeAllT $
+                        evalEnvUnifierT
+                            Not.notSimplifier
+                            (unificationProcedure SideCondition.top leftTerm rightTerm)
+
+                -- for each unification result, attempt to refute the formula
+                remainders ::
+                    [(OrPattern RewritingVariableName, Substitution RewritingVariableName)] <-
+                    mapM (checkUnifiedConsequent definedConfig) unified
+
+                let (successes, stucks) = partition (isBottom . fst) remainders
+
+                case (successes, stucks) of
+                    ((_bottom, subst) : _, _) ->
+                        -- success for at least one, return (ignore other failures)
+                        pure $ Implied (claimPattern, Just subst)
+                    ([], []) ->
+                        -- no successful unification, thus not implied
+                        pure $ NotImplied (claimPattern, Nothing)
+                    ([], stuck : _) ->
+                        -- successful unification but all stuck,
+                        -- return with first stuck term as antecedent
+                        let stuckClaim =
+                                (claimPattern :: ClaimPattern){left = OrPattern.toPattern sort (fst stuck)}
+                         in pure $ NotImpliedStuck (stuckClaim, Just $ snd stuck)
+  where
+    sort = termLikeSort leftTerm
+
+    claimPattern@ClaimPattern{left = renamedLeft, right = orRight, existentials} =
+        ClaimPattern.refreshExistentials $
+            -- ensures existentials of RHS do not occur free in LHS
+            ClaimPattern.mkClaimPattern inLeft (OrPattern.fromPattern inRight) inExistentials
```

Comment:
Not quite sure here, we should maybe rather _check_ that the input data does not produce name clashes when removing existentials, and return an informative error otherwise. That (new) error type might be appropriate for the `assertFunctionLikeConfiguration` above, too.